pds_doi_core/outputs/osti_web_client.py

```diff
@@ -73,7 +73,18 @@ def webclient_submit_existing_content(self, payload, i_url =None, i_username=Non
             n_records += 1
             result = {'doi': record.xpath('doi')[0].text,
                       'status': record.get('status')}
-            o_status[record.xpath('related_identifiers/related_identifier/identifier_value')[0].text] = result
+
+            # Some response from OSTI includes the 'doi_message' tag, fetch it.
+            # <doi_message>'Registration update failed: {"errors":[{"source":"publisher","title":"[facet \'minLength\'] The value has a length of \'0\'; this underruns the allowed minimum length of \'1\'. at line 14, column 0"}]}</doi_message>
+            if len(record.xpath('doi_message')) > 0:
+                result['doi_message'] = record.xpath('doi_message')[0].text
+
+            #o_status[record.xpath('related_identifiers/related_identifier/identifier_value')[0].text] = result
```

Comment:
Thanks for pointing that out. I think when I did that I was getting the lidvid identifier from the osti record. We should test carefully the action we were working on then (reserve and draft).

The specific case which we would need to test would be reserve with the submit flag = True.

Reply:
Sometimes the unit test does not reveal the bug depend on where the bug is.

I think we are good now.

Reply:
Now I think that the primary bug was that we missed to have the relatedIdentifier in the draft or reserverd result. This field contains the lidvid so it is very important for us (for PDS). To be on the safe side can you start by understanding why the relatedIdentifier is not produced in reserved or draft action ? Then we might want to restore this piece of code.

In the case of draft, the relatedIdentifier was missing from the mustache template (my bad). You need to add this piece of code in the mustache template:
    <related_identifiers>
            <related_identifier>
                <identifier_type>URL</identifier_type>
                <identifier_value>{{related_identifier}}</identifier_value>
                <relation_type>Cites</relation_type>
            </related_identifier>
        </related_identifiers>

(I took it from the reserve mustache template)

Reply:
I did that in branch #24_46_tl

Reply:
The snippet of code has been added to the mustache template for 'draft' action.

Reply:
Ok, thanks